pageserver/src/http/routes.rs

```diff
@@ -495,13 +495,39 @@ async fn get_lsn_by_timestamp_handler(
     let timeline = active_timeline_of_active_tenant(tenant_id, timeline_id).await?;
     let result = timeline.find_lsn_for_timestamp(timestamp_pg, &ctx).await?;
 
-    let result = match result {
-        LsnForTimestamp::Present(lsn) => format!("{lsn}"),
-        LsnForTimestamp::Future(_lsn) => "future".into(),
-        LsnForTimestamp::Past(_lsn) => "past".into(),
-        LsnForTimestamp::NoData(_lsn) => "nodata".into(),
-    };
-    json_response(StatusCode::OK, result)
+    if request
+        .headers()
+        .get(header::ACCEPT)
+        .map(|v| v == "application/json")
+        .unwrap_or_default()
+    {
+        #[derive(serde::Serialize)]
+        struct Result {
+            lsn: String,
+            r#type: &'static str,
+        }
+        let (lsn, type_) = match result {
+            LsnForTimestamp::Present(lsn) => (lsn, "present"),
+            LsnForTimestamp::Future(lsn) => (lsn, "future"),
```

Comment:
It's it possible to add a test for new `future` case? IIRC, we were always running `vacuum` after inserting data in both neon python tests and e2e tests

With this change we should be able to:
- insert data
- don't do any new commits
- get currest ts from compute
- get lsn from pageserver (should be future + lsn)
- create branch and see data there

Reply:
I don't change anything about the returned values, just the format. Therefore, maybe I can add those tests as a followup once the values also change?

Reply:
Yeah, up to you